### DISCUSSION TITLE
Installation : Found package configuration file: but it set FMS_FOUND to FALSE so package "FMS" is considered to be NOT   FOUND.

### DISCUSSION THREAD

**Discussion by ssk351** [ufs-community/ufs-weather-model#2276] [Q&A]
Hi,
I have tried to install the ufs using spack 
initially i have installed spack-stack. and I have tried ufs, then it showing the error
Machine: local machine with Ubuntu 22.04 
Found package configuration file:
   ..... path ...
  but it set FMS_FOUND to FALSE so package "FMS" is considered to be NOT
  FOUND.
Could please suggest the possible way to overcome the error

**Comment by natalie-perlin** [comment]
@ssk351 
Could you please specify the version of spack-stack you are using (checking out from GitHub), what are your compilers (modules), and exact steps you do to build the spack-stack? What are you compiler/MPI modules and versions, modules loaded when spack-stack is built, configuration in ./envs/[_your-custom-env_]/site/*.yaml and ./envs/[_your-custom-env_]/common/ *.yaml ? 

It is likely you could get more help and the quicker solution if you post this issue on https://github.com/JCSDA/spack-stack/discussions !..

**Reply by ssk351** [reply]
I am installing in HPC machine with centos. spack and gcc version are 0.17 and 12.0 respectively. 
module loaded were openmpi/4.4.0
I just used ufs-utils in the spack.

**Comment by gspetro-NOAA** [comment]
@ssk351 We're happy to help if you can provide some of the information @natalie-perlin requested. If you figured out the answer yourself, feel free to post it here for other users to benefit and/or let us know that we can close the discussion.

**Reply by ssk351** [reply]
Yes, I am working on it. I will update to you soon

**Reply by gspetro-NOAA** [reply]
@ssk351 How's it going? Do you have any further questions on this topic?

**Comment by ssk351** [comment]
Hi,
Sorry for the late reply!!
We could install the spack-stack, but I am not getting how to load these
packages.
How to load /install them?
Could you please suggest a possible way...
Thank you
Looking forward to hearing from you
$spack repo list (output: attached)
Regards
Saikrishna

**Reply by gspetro-NOAA** [reply]
@ssk351 In general, modules are loaded with a `module load name_of_module` statement. Sometimes it needs to be preceded by a module use statement such as `module use path/to/modulefiles`. I did reach out to our team that manages spack-stack, but some key SMEs are traveling at a conference or taking personal time off, so there may be a delay in reply if you need more information than that. I also noticed that in your post, it says `$spack repo list (output: attached)`, but there is no output attached. Could you attach the output so that when our SMEs are able to respond, they have the information they need? 

Thanks,
Gillian

**Comment by ssk351** [comment]
I am attaching the history that i have done.. Please help to resolve this
issue..
I have loaded spack/0.18, compiler/inter/2019 and cmake (using module load
...package name)
I have loded the packages like fms,emfs, etc using (spack load ...package
name).
Here, I am attaching the text document (includes the history/loaded
packages) and the error that i have got (attachment-2).
Thank you
#############
Regards
Saikrihna


On Thu, Jul 25, 2024 at 2:51 AM Gillian Petro ***@***.***>
wrote:

> @ssk351 <https://github.com/ssk351> In general, modules are loaded with a module
> load name_of_module statement. Sometimes it needs to be preceded by a
> module use statement such as module use path/to/modulefiles. I did reach
> out to our team that manages spack-stack, but some key SMEs are traveling
> at a conference or taking personal time off, so there may be a delay in
> reply if you need more information than that. I also noticed that in your
> post, it says $spack repo list (output: attached), but there is no output
> attached. Could you attach the output so that when our SMEs are able to
> respond, they have the information they need?
>
> Thanks,
> Gillian
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/discussions/2276#discussioncomment-10142511>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFTLE7MDBFOVGDAPIVEQTUDZOALD5AVCNFSM6AAAAABHQIZSSGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJUGI2TCMI>
> .
> You are receiving this because you were mentioned.Message ID:
> <ufs-community/ufs-weather-model/repo-discussions/2276/comments/10142511@
> github.com>
>

  spack load cmake/vptftol
  spack load fms
  spack load esmf/pcjidlz
  spack load bacio/am77vrq
  spack load g2/akn4ehe
  spack load g2c
  spack load geos
  spack load gfsio
  spack load grib_util
  spack load gsl/4ts72yb
  spack load hdf5/ybrks7r
  spack load ip/dbfesad
  spack load ip2
  spack load jasper/ywzgcy4
  spack load jpeg
  spack load landsfcutil
  spack load libpng/mbvrq3u
  spack load madis
  spack load mapl
  spack load nco
  spack load cdo+netcdf
  spack load upp
  spack load zlib/fejhazx
  spack load yafyaml
  spack load wgrib2
  spack load udunits/kcww3as
  spack load szip/6ihioep
  spack load sqlite/smfghxz
  spack load sp
  spack load sp/4vga7y3
  spack load sigio/o45qjv5
  spack load sfcio/o6eyija
  spack load proj/fqrkxig
  spack load nemsio/g2wks45
  spack load nemsiogfs
  spack load nemsi
  spack load pnetcdf
  spack load pnetcdf
  cd spack-stack-1.7.0/
  source setup.sh
  spack env activate envs/skylab-8.0.0/
 spack load ***@***.***/rxdnztn
  spack load ***@***.***
  spack load ***@***.***
  spack load ***@***.***/dayrkxm
  spack load sqlite/smfghxz
  spack load udunits/sncj47m
  spack load w3emc/4f6zqvs
  spack load w3nco/633itpg
  spack load sp/awqcy2l

**Comment by rickgrubin-noaa** [comment]
@ssk351:

Reviewing the comment history, there is a lot of context and information is missing here. 

As @gspetro-NOAA suggested, this discussion likely belongs in the [spack-stack github discussions](https://github.com/JCSDA/spack-stack/discussions); if you choose to not move the discussion, there, then @ssk351 please outline the following, and include it as an attachment (or multiple attachments) using the "**_Paste, drop, or click to add files_**" functionality provided as part of submitting comments. So far, only pieces of information you're intending to provide are showing up here (and not as attachments but rather as text within comments).

- the details of the host on which you're building; in particular, the output from the commands:
  - `uname -a`
  - `cat /etc/centos-release`

- how you obtained `spack-stack` software -- via [spack-stack github](https://github.com/JCSDA/spack-stack) or elsewhere (previously installed on your host)?

- is `module` software ([`Lmod`](https://lmod.readthedocs.io/en/latest/index.html)) installed on your host? 
  - what is the output of the command `module --config` ?
  - does the command `tclsh` drop you into a `Tcl` shell?
      - and if this drops you into a `Tcl` shell, what is the output of `info patchlevel` ?

- how you built your `spack stack`  components -- are you following [these instructions](https://spack-stack.readthedocs.io/en/latest/index.html#)?

**Comment by climbfuji** [comment]
Note that with spack-stack we do not support `spack load` - we generate modulefiles and load those.

**Comment by ssk351** [comment]
Hi, this was not the level 1-3 machine..
This was seperate machine. According to the document this machine will come
under level 3-4

**Reply by gspetro-NOAA** [reply]
Hi @ssk351 , 

I believe the approach should be the same regardless. For any new machine (at any level of support), the approach is to generate a modulefile for it and load that. You can use a copy of an existing modulefile and alter that as needed. Note, however, that Levels 1-4 support are machines that we have proactively determined we will support in some capacity. A machine that is not specifically listed is simply unsupported. We do what we can to assist, but there's no guarantee we can get things working when we don't have access to that type of system. To that end, it would be helpful to respond to @rickgrubin-noaa 's questions. 

Best,
Gillian

**Reply by rickgrubin-noaa** [reply]
> Hi, this was not the level 1-3 machine.. This was seperate machine. According to the document this machine will come under level 3-4

Understood, and that's fine.

That said, we cannot help you without more information, and as @gspetro-NOAA said, starting with the questions posed above will help.

Thanks.

**Comment by ssk351** [comment]
Ok, thank you.

**Comment by gspetro-NOAA** [comment]
@ssk351 I'm going to close this discussion since we haven't heard from you in a while or received the information we have requested to troubleshoot further. If you work on the modulefiles for your system or otherwise need assistance, feel free to open a new discussion, and provide us as much of the information from the [Q&A Posting Guidelines](https://github.com/ufs-community/ufs-weather-model/discussions/1897) as possible. 

Best,
Gillian Petro

**Comment by ssk351** [comment]
Yes, plz do the needful
